projects/packages/forms/src/contact-form/class-contact-form-endpoint.php

```diff
@@ -9,12 +9,441 @@
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$response = parent::prepare_item_for_response( $item, $request );
+		$data     = $response->get_data();
+		$fields   = $this->get_fields_for_response( $request );
+
+		$base_fields   = array(
+			'email_marketing_consent' => '',
+			'entry_title'             => '',
+			'entry_permalink'         => '',
+			'feedback_id'             => '',
+		);
+		$data_defaults = array(
+			'_feedback_author'       => '',
+			'_feedback_author_email' => '',
+			'_feedback_author_url'   => '',
+			'_feedback_all_fields'   => array(),
+			'_feedback_ip'           => '',
+			'_feedback_subject'      => '',
+		);
+
+		$feedback_data = array_merge(
+			$data_defaults,
+			\Automattic\Jetpack\Forms\ContactForm\Contact_Form_Plugin::parse_fields_from_content( $item->ID )
+		);
+
+		$all_fields = array_merge( $base_fields, $feedback_data['_feedback_all_fields'] );
+
+		$data['date'] = get_the_date( 'c', $data['id'] );
+		if ( rest_is_field_included( 'uid', $fields ) ) {
+			$data['uid'] = $all_fields['feedback_id'];
+		}
+		if ( rest_is_field_included( 'author_name', $fields ) ) {
+			$data['author_name'] = $feedback_data['_feedback_author'];
+		}
+		if ( rest_is_field_included( 'author_email', $fields ) ) {
+			$data['author_email'] = $feedback_data['_feedback_author_email'];
+		}
+		if ( rest_is_field_included( 'author_url', $fields ) ) {
+			$data['author_url'] = $feedback_data['_feedback_author_url'];
+		}
+		if ( rest_is_field_included( 'author_avatar', $fields ) ) {
+			$data['author_avatar'] = empty( $feedback_data['_feedback_author_email'] ) ? '' : get_avatar_url( $feedback_data['_feedback_author_email'] );
+		}
+		if ( rest_is_field_included( 'email_marketing_consent', $fields ) ) {
+			$data['email_marketing_consent'] = $all_fields['email_marketing_consent'];
+		}
+		if ( rest_is_field_included( 'ip', $fields ) ) {
+			$data['ip'] = $feedback_data['_feedback_ip'];
+		}
+		if ( rest_is_field_included( 'entry_title', $fields ) ) {
+			$data['entry_title'] = $all_fields['entry_title'];
+		}
+		if ( rest_is_field_included( 'entry_permalink', $fields ) ) {
+			$data['entry_permalink'] = $all_fields['entry_permalink'];
+		}
+		if ( rest_is_field_included( 'subject', $fields ) ) {
+			$data['subject'] = $data['_feedback_subject'];
+		}
+		if ( rest_is_field_included( 'fields', $fields ) ) {
+			$data['fields'] = array_diff_key(
+				$all_fields,
+				$base_fields
+			);
+		}
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Handles bulk actions for Jetpack Forms responses.
+	 *
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 *
+	 * @return WP_REST_Response A response object..
+	 */
+	public function bulk_actions( $request ) {
+		$action   = $request->get_param( 'action' );
+		$post_ids = $request->get_param( 'post_ids' );
+
+		if ( $action && ! is_array( $post_ids ) ) {
+			return new $this->error_response( __( 'Bad request', 'jetpack-forms' ), 400 );
```

Comment:
```suggestion
			return $this->error_response( __( 'Bad request', 'jetpack-forms' ), 400 );
```

This code calls a constructor on an object that has already been instantiated by error_response. This should result in a fatal error: https://3v4l.org/kKJYT.

Reply:
Good stuff here. Thanks! I've tried a few times to commit your suggestions through GH and it has some problems. I'll do it manually.

Also noting that especially the `bulk actions` code was just copied over and it will be good to check again to see what we can improve. 

One simple example is to see if we can remove some of the functions and handle some updates through REST API by updating the status etc..